tensorflow/python/ops/functional_ops.py

```diff
@@ -646,6 +646,207 @@ def compute(i, a_flat, tas):
+  of the result tensor is `[len(values)] + fn(initializer, values[0]).shape`.
+
+  This method also allows multi-arity `elems` and accumulator.  If `elems`
+  is a (possibly nested) list or tuple of tensors, then each of these tensors
+  must have a matching first (unpack) dimension.  The second argument of
+  `fn` must match the structure of `elems`.
+
+  If no `initializer` is provided, the output structure and dtypes of `fn`
+  are assumed to be the same as its input; and in this case, the first
+  argument of `fn` must match the structure of `elems`.
+
+  If an `initializer` is provided, then the output of `fn` must have the same
+  structure as `initializer`; and the first argument of `fn` must match
+  this structure.
+
+  For example, if `elems` is `(t1, [t2, t3])` and `initializer` is
+  `[i1, i2]` then an appropriate signature for `fn` in `python2` is:
+  `fn = lambda (acc_p1, acc_p2), (t1, [t2, t3]):` and `fn` must return a list,
+  `[acc_n1, acc_n2]`.  An alternative correct signature for `fn`, and the
+   one that works in `python3`, is:
+  `fn = lambda a, t:`, where `a` and `t` correspond to the input tuples.
+
+  Args:
+    fn: The callable to be performed.  It accepts two arguments.  The first
+      will have the same structure as `initializer` if one is provided,
+      otherwise it will have the same structure as `elems`.  The second
+      will have the same (possibly nested) structure as `elems`.  Its output
+      must have the same structure as `initializer` if one is provided,
+      otherwise it must have the same structure as `elems`.
+    elems: A tensor or (possibly nested) sequence of tensors, each of which
+      will be unpacked along their first dimension.  The nested sequence
+      of the resulting slices will be the first argument to `fn`.
+    initializer: (optional) A tensor or (possibly nested) sequence of tensors,
+      initial value for the accumulator, and the expected output type of `fn`.
+    parallel_iterations: (optional) The number of iterations allowed to run
+      in parallel.
+    back_prop: (optional) True enables support for back propagation.
+    swap_memory: (optional) True enables GPU-CPU memory swapping.
+    infer_shape: (optional) False disables tests for consistent output shapes.
+    name: (optional) Name prefix for the returned tensors.
+
+  Returns:
+    A tensor or (possibly nested) sequence of tensors.  Each tensor packs the
+    results of applying `fn` to tensors unpacked from `elems` along the first
+    dimension, and the previous accumulator value(s), from first to last.
+
+  Raises:
+    TypeError: if `fn` is not callable or the structure of the output of
+      `fn` and `initializer` do not match.
+    ValueError: if the lengths of the output of `fn` and `initializer`
+      do not match.
+
+  Examples:
+    ```python
+    elems = np.array([1, 2, 3, 4, 5, 6])
+    sum = scanr(lambda a, x: a + x, elems)
+    # sum == [720, 720, 360, 120, 30, 6]
```

Comment:
huh? Isn't that 6, 11, 15 etc?

Reply:
@drpngx Sorry about that. I was intending to show `multiply` in the example but ends up with `+`. The docstring has been updated now.